crates/cli/src/subcommands/sql.rs

```diff
@@ -72,50 +82,205 @@ pub(crate) async fn parse_req(mut config: Config, args: &ArgMatches) -> Result<C
             let mut table = stmt_result_to_table(stmt_result)?;
-            if with_stats {
-                // The `tabled::count_rows` add the header as a row, so subtract it.
-                let row_count = print_row_count(table.count_rows().wrapping_sub(1));
-                table.with(tabled::settings::panel::Footer::new(row_count));
+            if self.opts.with_row_count {
+                let rows = stmt_result.rows.len();
+                table.with(Footer::new(format!(
+                    "({} {})",
+                    rows,
+                    if rows == 1 { "row" } else { "rows" }
+                )));
+            }
+            out.write_all(table.to_string().as_bytes()).await?;
+            if let Some(sep) = sep {
+                out.write_all(sep).await?;
+            } else {
+                out.write_all(b"\n").await?;
+            }
+        }
+
+        if self.opts.with_timing {
+            out.write_all(self.fmt_timing().as_bytes()).await?;
+        }
+        out.flush().await?;
+        Ok(())
+    }
+
+    /// Renders each query result as a sequence of CSV-formatted lines.
+    ///
+    /// The column names are printed as a comment (a line starting with '#') at
+    /// the beginning of each row.
+    ///
+    /// # Example
+    ///
+    /// A query `SELECT * FROM Message; SELECT * FROM Users;` against the
+    /// `quickstart-chat` application may return the following:
+    ///
+    /// ```text
+    /// # sender,sent,text
+    /// 0xaba52919637a60eb336e76eed40843653bfb3d9d94881d78ab13dda56363b993,1718611618273455,hello
+    /// # identity,name,online
+    /// 0xaba52919637a60eb336e76eed40843653bfb3d9d94881d78ab13dda56363b993,(none = ()),true
+    /// ```
+    async fn render_csv(self, mut out: impl AsyncWrite + Unpin) -> anyhow::Result<()> {
+        for stmt_result in &self.stmt_results {
+            let mut csv = csv_async::AsyncWriter::from_writer(&mut out);
+            let StmtResultJson { schema, rows } = stmt_result;
+            let ts = Typespace::EMPTY.with_type(schema);
+
+            // Render column names as a comment, e.g.
+            // `# sender,sent,text`
+            for (i, name) in schema.names().enumerate() {
+                if i == 0 {
+                    csv.write_field(format!("# {}", name)).await?;
+                } else {
+                    csv.write_field(&*name).await?;
+                }
             }
-            anyhow::Ok(table)
-        })
-        .process_results(|it| println!("{}", it.format("\n\n")))?;
-    if with_stats {
-        print_timings(now);
+            // Terminate record.
+            csv.write_record(None::<&[u8]>).await?;
+
+            for row in rows {
+                let row = from_json_seed(row.get(), SeedWrapper(ts))?;
+                for field in fmt_row_psql(&row, ts) {
+                    // Remove quotes around string values to prevent quoting.
+                    csv.write_field(field.trim_matches('"')).await?;
```

Comment:
Also, I think is better to use quotes for strings. Like anything with CSV there are no strict rules, but certainly is a good practice.

This helps in these common scenarios:

* A string has a comma inside
* A numeric value can too: `20,2`
* Line breaks trip parsers, most I know can read well instead of "hello\nworld"
* Parsing a CSV always leads to the question about the commas, with quotation this gets 'solved' (however leads to escaping of the quotation itself)

Reply:
Oh, no, you're misreading this: the CSV library does quote strings, but the PsqlWrapper does, too. So we end up with double-double-quoted strings. I should reword the comment as ".. prevent double double-quoting".